mono/eglib/gpath.c

```diff
@@ -251,11 +251,17 @@ g_find_program_in_path (const gchar *program)
 		
 		x = NULL;
 		probe_path = g_build_path (G_DIR_SEPARATOR_S, l, program, NULL);
-		if (access (probe_path, X_OK) == 0){ /* FIXME: on windows this is just a read permissions test */
+#if !defined(NO_HAVE_ACCESS)
```

Comment:
Where does this come from?

Reply:
I added `NO_HAVE_ACCESS` to handle platforms that don't have `access` in their API. For most platforms, it is not defined, but we do define it in our build for platforms which don't have this API.

Reply:
This should be integrated into mono using a configure check. Something like this:
```
	dnl **********************************
	dnl *** Check for getaddrinfo ***
	dnl **********************************
	AC_MSG_CHECKING(for getaddrinfo)
		AC_TRY_LINK([
		#include <unistdc.h>
	], [
		access (NULL, 0);
	], [
		# Yes, we have it...
		AC_MSG_RESULT(yes)
		AC_DEFINE(HAVE_ACCESS, 1, [Have access])
	], [
		AC_MSG_RESULT(no)
	])
```

The result would then define the a HAVE_ACCESS variable if it's what we need to use.

Reply:
I've added this configure check at b41ac55c94d3ae5120a56312f07de5022c28c245.